hdf/test/file1.c

```diff
@@ -36,16 +36,17 @@ test_file_limits(void)
     MESSAGE(5, puts("Opening many files of same name"););
     for (i = 0; i < BIG; i++) {
         files[i] = Hopen("thf.hdf", DFACC_RDWR, 0);
+        CHECK_VOID(files[i], -1, "Hopen");
         if (files[i] < 0) {
-            /*            i++; */
             break;
-        } /* end if */
+        }
     }
     MESSAGE(5, printf("Opening stopped at %d/%d files\n", i, BIG););
 
     MESSAGE(5, puts("Closing all files"););
-    for (i--; i >= 0; i--) {
+    for (i = i - 1; i >= 0; i--) {
         ret = Hclose(files[i]);
+        CHECK_VOID(ret, -1, "Hclose");
```

Comment:
It's probably a good idea for tests to check for errors instead of swallowing them...